### PR TITLE
Refactor set response object

### PR DESCRIPTION
**Pull request by doktordirk**

### CODE REVIEW

**Comment by doktordirk** [comment]
closes https://github.com/SpoonX/aurelia-authentication/issues/120

**Comment by AdamWillden** [comment]
This is a breaking change for me I think. I can make the change no problem but just to let you know

**Comment by doktordirk** [comment]
@AdamWillden what's the breaking part

**Comment by AdamWillden** [comment]
It is strictly because I needed this feature before it was officially
implemented therefore I was reaching through with
`authService.authorisation.reaponseObject = newResponse` or whatever it
is. Because the method wasn't part of the officialy supported API it's not
really breaking I guess

**Comment by doktordirk** [comment]
@AdamWillden i re-added the getter/setter. any chance you +1 me on this ;) i know i should go back and forth so ofter, but the get/set was stupid in the first place (i think i needed it in some step, but didn't consider to replace them then). anyways, this way it's better for a bc setTimeout approach (next PR)

**Comment by AdamWillden** [comment]
I'll +1 anything :-p Are you saying I won't need to change anything now then? I don't mind if it's better either way

**Comment by doktordirk** [comment]
you woudn't need to change anything. just if the timeout PR gets in 3.1, i'd recommend using authService.setResponseObject(response) to use the non dirty checked authService.authenticated

File: src/storage.js [hunk 1]
```diff
@@ -16,13 +16,13 @@ export class Storage {
 
   set(key, value) {
     if (window[this.config.storage]) {
-      return window[this.config.storage].setItem(key, value);
```
Comment (RWOverdijk) [inline src/storage.js]: Why did you remove the return?

File: src/authService.js [hunk 2]
```diff
@@ -7,9 +7,35 @@ import {BaseConfig} from './baseConfig';
 
 @inject(Authentication, BaseConfig)
 export class AuthService {
+  /**
+   * The Authetnication instance that handles the token
```
Comment (RWOverdijk) [inline src/authService.js]: Authentication

File: src/authService.js [hunk 3]
```diff
@@ -7,9 +7,35 @@ import {BaseConfig} from './baseConfig';
...
     this.authentication = authentication;
     this.config         = config;
+
+    // get token stored in previous format over
+    const oldStorageKey = config.tokenPrefix
+                        ? config.tokenPrefix + '_' + config.tokenName
+                        : config.tokenName;
+    const oldToken = authentication.storage.get(oldStorageKey);
+
+    if (oldToken) {
```
Comment (RWOverdijk) [inline src/authService.js:32]: What is this for?

**Comment by RWOverdijk** [comment]
I'm curious as to the purpose of this.

File: src/authService.js (hunk 3 above)
Comment (doktordirk) [inline src/authService.js:32]: we changed the key and format of the storage. so, in case an enduser, is logged in (old format) but now runs the updated app requiering new format, it gets little messy. that code just looks for the old format and resaves it in the new one so the enduser won't notice something has changes

File: src/storage.js (hunk 1 above)
Comment (doktordirk) [inline src/storage.js]: since it isn't returning anything (storage specs)

File: src/authService.js (hunk 2 above)
Comment (doktordirk) [inline src/authService.js]: ah no actually i mean autoethnication, so the process of automatically adapting an ethnic identity :smile:

File: src/authService.js (hunk 3 above)
Comment (RWOverdijk) [inline src/authService.js:32]: Looks icky haha. Thanks for your explanation

File: src/authService.js (hunk 2 above)
Comment (RWOverdijk) [inline src/authService.js]: _slowly claps_ well done

File: src/storage.js (hunk 1 above)
Comment (RWOverdijk) [inline src/storage.js]: Fair enough

File: src/authService.js (hunk 3 above)
Comment (doktordirk) [inline src/authService.js:32]: i have a test for this in actually. (and had it tested in my app) so, i'm quite certain

**Comment by doktordirk** [comment]
you mean the whole thing? for one get/set was stupid of me. secondly, makes the timeout thing possible

**Comment by RWOverdijk** [comment]
👍
